src/bootstrap/lib.rs

```diff
@@ -892,7 +893,12 @@ impl Build {
 
     /// Returns the sysroot of the snapshot compiler.
     fn rustc_snapshot_sysroot(&self) -> &Path {
-        self.initial_rustc.parent().unwrap().parent().unwrap()
+        static SYSROOT_CACHE: OnceCell<PathBuf> = once_cell::sync::OnceCell::new();
+        SYSROOT_CACHE.get_or_init(|| {
+            let mut rustc = Command::new(&self.initial_rustc);
+            rustc.args(&["--print", "sysroot"]);
+            output(&mut rustc).into()
```

Comment:
I think we need a .trim() here -- at least for me, it looks like --print sysroot has a trailing newline at the end:

```
$ rustc --print sysroot | xxd
00000000: 2f68 6f6d 652f 6d61 726b 2f2e 7275 7374  /home/mark/.rust
00000010: 7570 2f74 6f6f 6c63 6861 696e 732f 6e69  up/toolchains/ni
00000020: 6768 746c 792d 7838 365f 3634 2d75 6e6b  ghtly-x86_64-unk
00000030: 6e6f 776e 2d6c 696e 7578 2d67 6e75 0a    nown-linux-gnu.
```